src/main/java/org/spongepowered/api/data/type/Material.java

```diff
@@ -22,17 +22,19 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.data.manipulator.immutable.block;
+package org.spongepowered.api.data.type;
 
-import org.spongepowered.api.block.BlockTypes;
-import org.spongepowered.api.data.manipulator.immutable.ImmutableVariantData;
-import org.spongepowered.api.data.manipulator.mutable.block.QuartzData;
-import org.spongepowered.api.data.type.QuartzType;
+import org.spongepowered.api.CatalogType;
 
-/**
- * An {@link ImmutableVariantData} for the {@link QuartzType}. Usually
- * applicable to {@link BlockTypes#QUARTZ_BLOCK}.
- */
-public interface ImmutableQuartzData extends ImmutableVariantData<QuartzType, ImmutableQuartzData, QuartzData> {
+public interface Material extends CatalogType {
```

Comment:
After thinking about it more today, I don't like the idea of naming it `Material`. Maybe call it `Variety` and then `Varieties.OAK` and `Varieties.QUARTZ` could then work. This way you can also have `Varieties.STAIR` or `PLANK` even. This way the "type" is not defined as a Material (since stair is not a material, but a variety of a block). This also allows for us to make a combination of varieties for a specific object to "describe" it.

Reply:
Thinking on it more, `Variety` could even be introduced in API 7 (given some time this week maybe to solidify it as a possibility), in which a "moving" towards it is somewhat available before the drastic change of some of this data removal.